domain/access/state/permission.go

```diff
@@ -92,34 +92,41 @@ type AddUserPermissionArgs struct {
 }
 
 // AddUserPermission adds a permission for the given user on the given target.
+// Validates that the target exists and that the Access level is appropriate
+// for the target before insert.
 func AddUserPermission(ctx context.Context, tx *sqlair.TX, spec AddUserPermissionArgs) error {
+	// Validate the access is appropriate for the target.
+	if err := spec.Target.ValidateAccess(spec.Access); err != nil {
+		return fmt.Errorf("%q for %q %w ", spec.Access, spec.Target.Key, accesserrors.PermissionAccessInvalid)
+	}
+	// Validate the target exists.
+	if err := targetExists(ctx, tx, spec.Target.Key); err != nil {
+		return errors.Trace(err)
+	}
+
 	// Insert a permission doc with
 	// * permissionObjectAccess as permission_type_id
 	// * uuid of the user (spec.User) as grant_to
 	// * spec.Target.Key as grant_on
 	newPermission := `
-INSERT INTO permission (*)
-VALUES ($dbAddUserPermission.*)
+INSERT INTO permission (uuid, permission_type_id, grant_to, grant_on)
+SELECT $dbAddUserPermission.uuid, t.id AS permission_type_id, u.uuid, $dbAddUserPermission.grant_on
+FROM   v_user_auth u, permission_access_type t
+WHERE  u.uuid = $dbAddUserPermission.grant_to
+--AND    u.disabled = false
```

Comment:
I have concerns here. The user row in the `user_authentication` table isn't added until a password is set, or the activation key is used. The result is that we cannot verify if the user is disabled before adding a permission. 

This is a change in behavior from 3.x